File: Algorithm/Code_Materials/c_or_cpp/charpter8_tree_hot_problems/level1/merge_trees.cpp
#include <vector>
#include <stack>
#include <iostream>
using namespace std;

struct TreeNode
{
    int val;
    TreeNode *left;
    TreeNode *right;
    TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
};
 
 TreeNode* mergeTrees(TreeNode* t1, TreeNode* t2) {
        if (t1 == nullptr) {
            return t2;
        }
        if (t2 == nullptr) {
            return t1;
        }
        auto merged = new TreeNode(t1->val + t2->val);
        merged->left = mergeTrees(t1->left, t2->left);
        merged->right = mergeTrees(t1->right, t2->right);
        return merged;
    }


int main() {
    // 创建测试用例的二叉树
    TreeNode* root1 = new TreeNode(1);
    root1->left = new TreeNode(2);
    root1->right = new TreeNode(3);
    root1->left->left = new TreeNode(4);
    root1->left->right = new TreeNode(5);
    root1->right->left = new TreeNode(6);
    root1->right->right = new TreeNode(7);
    TreeNode* root2 = new TreeNode(1);
    root2->left = new TreeNode(12);
    root2->right = new TreeNode(13);

     
    TreeNode* tree=mergeTrees(root1, root2);   // 输出1，表示两个二叉树相同
    return 0;
}

File: Algorithm/Code_Materials/python_source/charpter8_tree_hot_problems/level2/topic_双指针/merge_trees.py
# coding=utf-8
from python_source.tool.Node import Node
from python_source.tool.Tree import Tree

#  判断树是否一样
class MergeTrees:
    def mergeTrees(self, t1, t2):
        if not t1:
            return t2
        if not t2:
            return t1

        merged = Node(t1.val + t2.val)
        merged.left = self.mergeTrees(t1.left, t2.left)
        merged.right = self.mergeTrees(t1.right, t2.right)
        return merged


if __name__ == "__main__":
    tree = Tree()
    q = tree.init_tree_for_merge1()
    p = tree.init_tree_for_merge2()
    mergeTrees = MergeTrees()
    res = mergeTrees.mergeTrees(q, p)
    print res
